ext-apiserver/pkg/controller/machineset/controller.go

```diff
@@ -0,0 +1,52 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package machineset
+
+import (
+	"github.com/golang/glog"
+	"github.com/kubernetes-incubator/apiserver-builder/pkg/builders"
+
+	"k8s.io/kube-deploy/ext-apiserver/pkg/apis/cluster/v1alpha1"
+	listers "k8s.io/kube-deploy/ext-apiserver/pkg/client/listers_generated/cluster/v1alpha1"
+	"k8s.io/kube-deploy/ext-apiserver/pkg/controller/sharedinformers"
+)
+
+// +controller:group=cluster,version=v1alpha1,kind=MachineSet,resource=machinesets
+type MachineSetControllerImpl struct {
```

Comment:
Is this generated?
Or is this the final place for the MachineSet controller?

As https://github.com/kubernetes/kube-deploy/pull/620 is targeting another directory

Reply:
This will be the final place for machineset controller. #620 will need to refactored into the new directory. Please refer to machine controller as example.